students/Josh_HOff/lesson06/test_mailroom4.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+import mailroom4
+from mailroom4 import check_input
+from mailroom4 import check_donation
+from mailroom4 import create_report
+from mailroom4 import letters_to_everyone
+
+class MyTests(unittest.TestCase):
+
+    def setUp(self):
+        pass
+    def test_1(self):
+        self.assertFalse(check_input('quit'))
+    def test_2(self):
+        self.assertTrue(check_input('list'))
+    def test_3(self):
+        self.assertIsNone(check_input('New Donor'))
+    def test_4(self):
+        self.assertFalse(check_donation('New Donor', 'quit'))
+    def test_5(self):
+        self.assertTrue(check_donation('New Donor', 'Non-number'))
+    def test_6(self):
+        self.assertIsNone(check_donation('Ralph Anders', 300))
+        self.assertIn(300, mailroom4.donors_list['Ralph Anders'])
+    def test_7(self):
+        self.assertIsNone(check_donation('New Donor', 400.56))
+        self.assertIn(400.56, mailroom4.donors_list['New Donor'])
+    def test_8(self):        
+        self.assertEqual(create_report(), (mailroom4.top + mailroom4.rows))
```

Comment:
it looks like your `create_report` function returns the report data, you should use that instead of a global module variable that magically gets set.

Reply:
Okay, I adjusted the code to self.assertEqual(create_report(), mailroom4.reportvariable), is that what you're suggesting?

Reply:
in your test you should have one thing that executes your code and another thing that compares to what is expected which is usually defined inside of your test.

For example:
```
def test_create_report():
    expected = "some expected string"
    actual = create_report()
    self.assertEquals(actual, expected)
```

Reply:
Okay, I understand now. I've adjusted my code accordingly.